lms/static/scripts/frontend_apps/utils/format.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * Truncate a URL to be `maxLength` or fewer characters.
+ *
+ * If `url` needs to be shortened, `truncateURL` tries to preserve the most
+ * informative parts of the URL using some simple heuristics.
+ *
+ * If `url` is not a valid URL, it is just shortened to the first `maxLength - 1`
+ * characters.
+ *
+ * @param {string} url
```

Comment:
I think I like that this function takes a `string` instead of a `URL` object...it's more flexible for sure.